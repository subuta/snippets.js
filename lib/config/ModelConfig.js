import check from 'lib/_utils/apiCheck'

const ModelConfigShape = check.shape({
  schema: check.object
})

// eg:
// config = {
//   book: {
//     schema: {}
//   }
// }
export const validateModels = function (config) {
  return check.throw(check.objectOf(ModelConfigShape), config)
}

// eg:
// config = {
//   schema: {}
// }
export const validateModel = function (config) {
  return check.throw(ModelConfigShape, config)
}
