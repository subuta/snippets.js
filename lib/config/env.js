import { build, format, snippets as s } from 'bld.js'
import _ from 'lodash'

export default (imports = []) => {
  const vars = _.zipObject(imports, imports)

  return build`
    // load .env first.
    require('dotenv').config()
    
    const ${s.raw(vars)} = process.env
    
    export ${s.raw(vars)}
    
    export default ${s.raw(vars)}
  `
}
