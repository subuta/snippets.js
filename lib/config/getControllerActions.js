import _ from 'lodash'

export default (config = {}) => {
  const {
    only, // the routes will generated.
    except  // the routes will ignored.
  } = config

  const routes = [
    'index',
    'show',
    'create',
    'update',
    'destroy'
  ]

  if (only) {
    return _.intersection(routes, only)
  } else if (except) {
    return _.difference(routes, except)
  }

  return routes
}
