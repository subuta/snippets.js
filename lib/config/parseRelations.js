import _ from 'lodash'
import pluralize from 'pluralize'

export default (config = {}) => {
  const { schema } = config

  const relations = _.transform(schema.relations, (result, value, key) => {
    let relatedModel = null;
    if (value.belongsTo) {
      relatedModel = pluralize.singular(_.toLower(value.belongsTo))
    } else if (value.hasOne) {
      relatedModel = pluralize.singular(_.toLower(value.hasOne))
    } else if (value.hasMany) {
      relatedModel = pluralize(_.toLower(value.hasMany))
    } else if (value.hasAndBelongsToMany) {
      relatedModel = pluralize(_.toLower(value.hasAndBelongsToMany))
    }
    result[key] = relatedModel
  }, {})

  return relations
}
