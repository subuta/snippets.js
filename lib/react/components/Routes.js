import { build, format, snippets as s } from 'bld.js'
import _ from 'lodash'

export default () => {
  const imports = s.import([
    ['react', 'React'],
    ['_', 'lodash'],
    ['react-router-redux', null, [
      'ConnectedRouter'
    ]],
    ['src/utils/history', 'history']
  ])

  return build`
    ${imports}
    
    /* mat Custom imports [start] */
    /* mat Custom imports [end] */
    
    let routes = null
    
    /* mat Custom routes [start] */
    /* mat Custom routes [end] */
    
    export default routes

  `
}