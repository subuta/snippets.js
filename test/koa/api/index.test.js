/* global expect, describe, it */

import { build, format, snippets as s } from 'bld.js'

import Routes from 'lib/koa/api'

import { Routes as RoutesConfig } from 'test/fixtures/config'

describe('knex/api/index', () => {
  it('should create koa Base', () => {
    expect(format(Routes(RoutesConfig))).toMatchSnapshot()
  })
})
