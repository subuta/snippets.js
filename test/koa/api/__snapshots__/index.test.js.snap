// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`knex/api/index should create koa Base 1`] = `
"import Router from 'koa-router'
import _ from 'lodash'
import koaBody from 'koa-body'
import auth, {getCurrentUser} from './middlewares/auth'
import models from './middlewares/models'
import channel from './channel'
import comment from './comment'
import attachment from './attachment'
import user from './user'

const api = new Router({
  prefix: '/api'
})

// register routers to api.
const registerRouters = (routers) => {
  _.each(routers, (router, name) => {
    router.register && router.register(routers)
    api.use(router.routes())
  })
}

// routers set before auth middleware will not be protected
// parse body
api.use(
  koaBody({
    multipart: true
  })
)

// set jwt middleware
api.use(auth)

// inject Objection.js models middleware.
api.use(models)

// inject getCurrentUser to state for ease of use.
api.use(getCurrentUser)

// routers set after auth middleware will be protected
registerRouters({
  channel,
  comment,
  attachment,
  user
})

export default api
"
`;
